snapshot_dbg_extension/src/logLevelPicker.ts

```diff
@@ -17,6 +21,27 @@ export async function pickLogLevel(): Promise<string | undefined> {
             {'label': `${level} $(${level.toLowerCase()})`}
         ));
 
-    const selection = await vscode.window.showQuickPick(items, {'title': 'Select Log Level'});
+    // If we have one call to showQuickPick outstanding, any other calls to it
+    // during this time will immediately return with a value of undefined.
+    // During initialization time it's possible we'll be prompting the user
+    // to select a log level for logpoints found in the IDE in multiple
+    // different files. This can lead to this situation occurring, so here
+    // we serialize the calls.
+    while (gPickPending) {
+        await (sleep(250));
+    }
+
+    gPickPending = true;
+    const selection = await vscode.window.showQuickPick(items, {title});
+    gPickPending = false;
+
     return selection?.label.split(' ')[0].trim();
 }
+
+export async function pickLogLevelNewlyCreated(): Promise<string | undefined> {
+    return pickLogLevel('Select Log Level');
+}
+
+export async function pickLogLevelSyncedFromIDE(path: string, line: number): Promise<string | undefined> {
+    return pickLogLevel(`Select Log Level For Logpoint Being Synced From the IDE: ${path}:${line}`);
```

Comment:
This is pretty verbose.  We can probably remove "Being Synced From the IDE".

Reply:
Good call.